### PR TITLE
[cmake][Issue #341] Revert cuda_gen_all_arch flag and automatic build target selection

### PR DESCRIPTION
**Pull request by atadkase**
WAR fix for Issue #341 while performance drop debug is underway.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -70,14 +70,10 @@ To build Clara Genomics Analysis -
 ```bash
 mkdir build
 cd build
-cmake .. -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=install -Dcga_cuda_gen_all_arch=OFF
+cmake .. -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=install
```
Comment (r-mafi) [inline README.md:73]: if decided to remove -Dcga_cuda_gen_all_arch=OFF altogether for cudapoa, cudaaligner as well as cudamapper, then README documentation needs to be updated as well.

**Comment by tijyojwad** [comment]
@atadkase - let's benchmark `cudamapper` performance before pushing this change in. could be that we want to keep it for cudamapper and revert only for cudapoa/cudaaligner

**Comment by atadkase** [comment]
It seems that cudamapper perf degrades with this change. I'll rework it so that this affects only cudapoa and cudaaligner

**Comment by atadkase** [comment]
Closing this PR as #409 was opened to address the perf decrease in cudamapper with this change.

**Pull request closed without merging**